src/main/java/pl/edu/mimuw/BigInt.java

```diff
@@ -27,16 +44,242 @@ public BigInt neg() {
+            d[0] = 0;
+            result = new BigInt(d, true);
+          }
+          else //substracting one from another
+          {
+
+              int [] bigger = this.digits, smaller = this.digits;
+              boolean sign = true;
+              if(which_bigger == 1)
+              {
+                sign = this.isPositive;
+                bigger = this.digits;
+                smaller = other.digits;
+              }
+              else
+              {
+                sign = other.isPositive;
+                bigger = other.digits;
+                smaller = this.digits;
+              }
+              int [] smaller2 = new int [bigger.length];
+              int numer = smaller2.length-1;
+              //increasing the smaller number with leading 0
+              for(int i=smaller.length-1; i>=0; i--)
+              {
+                smaller2[numer] = smaller[i];
+                numer--;
+              }
+              while(numer >= 0)
+              {
+                smaller2[numer] = 0;
+                numer --;
+              }
+
+              //substracion
+
+              for(int i = bigger.length-1; i>=0; i--)
+              {
+                  if(bigger[i] >= smaller2[i])
+                  {
+                    bigger[i] -= smaller2[i];
+                  }
+                  else //finding non zero on the left
+                  {
+                    int where = 0;
+                    for(int j=i-1; j>=0; j--)
+                    {
+                        if(bigger[j] != 0)
+                        {
+                          where = j;
+                          break;
+                        }
+                    }
+                    bigger[where] --;
+                    where--;
+                    while(where > i)
+                    {
+                      bigger[where] = 9;
+                      where--;
+                    }
+                    bigger[i] += 10;
+                    bigger[i] -= smaller2[i];
+                  }
+              }
+
+              //getting rid of leading 0
+               int nr = 0;
+
+            size = bigger.length;
+              for(int i=0; i<size; i++)
+            {
+              if(bigger[i]!=0)
+                break;
+              else
+              nr++;
+            }
+            int c2[] = new int[size-nr];
+            int j = 0;
+            // rewriting to second arr
+
+            for(int i=nr; i<size; i++)
+            {
+              c2[j] = bigger[i];
+              j++;
+            }
+            result = new BigInt (other.digits, true);
+
+            result = new BigInt(c2, sign);
+          }
+      }
+      return result;
   }
 
   public BigInt times(BigInt other) {
-    throw new IllegalStateException("TODO task 3: implement multiplication");
+
+    boolean sign = false;
+    if (this.isPositive == other.isPositive)
+      sign = true;
+    int[] c = new int[this.digits.length + other.digits.length+1];
+    int where = 0;
```

Comment:
where what? xd nazwywajmy rzeczy op imieinu ;) to samo tyczy sie tych tablic `c`